### ISSUE TITLE
Feature Request: Add tooling to open database in local client (SequelPro, HeidiSQL, etc.)

### ISSUE THREAD

**Issue by dalin-** [lando/lando#1168]
# User Story

Lando users can easily open the database in their local MySQL client of choice

# Acceptance Criteria

As a Lando user who works with a project that uses MySQL
If I type `lando pancakes` 
Then the database is opened in whatever MySQL client I have installed (SequelPro, HeidiSQL, MySQL Workbench, MySQL CLI)

# Rationale

Currently if you want to do this there are several steps involved:

1. `lando info`
2. Scan through the JSON output to find the database db/user/pw/host/port, but make sure that you're looking at the external credentials, the internal ones won't work.
3. Open your client of choice.  If you've saved the credentials that you last used, then you should only have to change the port, otherwise you'll need to enter all 5.

This plugin exists for Terminus:
https://github.com/terminus-plugin-project/terminus-pancakes-plugin

This feature also exists for DDEV:
https://ddev.readthedocs.io/en/latest/users/developer-tools/#database-management

**Comment by tylerssn** [comment]
This would be convenient.

My current workaround is to modify my `.lando.yml` so that the mysql service has a static port forwarded by using: `portforward: 33080 # Or, some other non-sudo-requiring port`. Changing the port does require a rebuild but your database *should* persist across rebuilds. This setup allowed my IDE to connect without having to be reconfigured.

The second option is to enable PHPMyAdmin within Lando.

Pending acceptance of PR https://github.com/lando/lando/pull/1169, I'm hoping to make a plugin that makes fetching service info more efficient either by exporting service info to the host environment OR a command such as `lando service {service_name} --port`

**Comment by dalin-** [comment]
Would love to see the wontfix label removed from this.  I think this is an important strategy piece that will help Lando compete with the other Docker alternatives.

**Comment by chrisarcherhydrant** [comment]
This would be a great feature and make managing DB's much easier

**Comment by mikemuller** [comment]
Docksal has this and make it much more convenient/efficient to work with local databases.

**Comment by benedikt-buchert** [comment]
https://stackoverflow.com/questions/26062195/open-a-sequel-pro-connection-from-command-line

**Comment by pirog** [comment]
I certainly understand the convenience of something like this and i'm all for making things easier.

That said this does break one of the fundamental design goals of Lando which is to containerize everything and be as uncoupled as possible from the host. Because i am not really a purist when it comes to those sort of considerations ill point to more practical concerns: this does not seem like a particularly scalable, easy to maintain (eg consistently reliable), or broad solution (eg lando is not a mysql/php only thing). It makes sense that terminus or ddev or docksal support something like this because of their narrower scope but im not sure it makes sense for lando.

A possible alternative here that _would_ make more sense to me is if there was some sort of easy user-configurable way to do this, that seems like it would alleviate a lot of the burden placed on Lando and be more scalable as tools change, new tools come online, Lando expands into other tech etc. I think the technical underpinning of something like that would probably be a way to run lando tooling commands directly on the host machine and have various ENVARS available for those commands eg the connection info for a database, etc. 

Anyway, those are my thoughts! if anyone has alternative implementations or riffs on above that also address the concerns expressed on the lando side def interested to hear them!

**Comment by tylerssn** [comment]
What? Did I hear @pirog say something about user plugins? /sarcasm 🙂

**Comment by pirog** [comment]
If that was a joke it went over my head

**Comment by tylerssn** [comment]
Just a call back to being able to build Lando plugins so that users could extend Lando functionality. My understanding is that functionality was shelved or perhaps never intended to be available. I tried with humor to resurface that conversation, using this issue as leverage.

**Comment by pirog** [comment]
That functionality is and has been available since the beginning. There is
even documentation and examples on how to use it. Lando itself uses it.

Regardless, my proposal above is not a suggestion that users create their
own plugins to accomplish this, it’s merely that they should have to
configure this on their own, similar to tooling. So, this probably isn’t a
good issue to use as leverage for whatever you think is lacking.

**Comment by tanc** [comment]
I had a spare morning so I wrote this plugin: https://github.com/tanc/lando-db-tools

> The intention of the plugin is to make working with Lando database containers and external tools a little easier. The plugin comes with three Lando 'tasks':

> 1. `lando dbport` - prints the external port and copies it to the clipboard
> 2. `lando workbench` - opens a connection using the MySQL Workbench GUI
> 3. `lando dbeaver` - opens a connection using the dbeaver GUI

I haven't tested this in any other environments other than my own Arch Linux. I've only added two GUI tools, both open source and cross platform. I imagine anything else that can take query parameters could be added, I'm open to pull requests.

Once #2502 is done it should be a little easier to install.

**Comment by tanc** [comment]
Oh and I should add the plugin is very MySQL/MariaDB centric at the moment
